spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/HttpServerStub.java

```diff
@@ -12,25 +12,47 @@
 public interface HttpServerStub {
 
 	/**
-	 * Port on which the server is running
+	 * Port on which the server is running. Return {@code -1} if not applicable.
 	 */
 	int port();
 
+	/**
+	 * Https port on which the server is running. Return {@code -1} if not applicable.
+	 */
+	default int httpsPort() {
+		return -1;
+	}
+
 	/**
 	 * Returns {@code true} if the server is running
 	 */
 	boolean isRunning();
 
 	/**
 	 * Starts the server on a random port. Should return itself to allow chaining.
+	 * @deprecated use {@link HttpServerStub#start(HttpServerStubConfiguration)}
 	 */
+	@Deprecated
 	HttpServerStub start();
 
 	/**
 	 * Starts the server on a given port. Should return itself to allow chaining.
+	 * @deprecated use {@link HttpServerStub#start(HttpServerStubConfiguration)}
 	 */
+	@Deprecated
 	HttpServerStub start(int port);
 
+	/**
+	 * Starts the server. Should return itself to allow chaining.
+	 * @param configuration - setup for the given stub
+	 */
+	default HttpServerStub start(HttpServerStubConfiguration configuration) {
+		if (configuration.isRandomPort()) {
+			return start();
```

Comment:
should we use deprecated method in the most current implementation?

Reply:
Yeah that's the default implementation - we don't want to break the compatibility.